src/main/java/org/jabref/gui/groups/GroupAddRemoveDialog.java

```diff
@@ -166,7 +167,8 @@ private boolean doAddOrRemove() {
             GroupTreeNodeViewModel node = (GroupTreeNodeViewModel) path.getLastPathComponent();
             if (checkGroupEnable(node)) {
 
-                List<BibEntry> entries = Globals.stateManager.getSelectedEntries();
+                List<BibEntry> entries = new ArrayList<>(Globals.stateManager.getSelectedEntries()); //we need to copy the contents of the oversable list here, because when remove is called,
```

Comment:
Please move the comment in front of the statement and fix the spelling (and maybe clarify the root for this change a bit more)